mbed_host_tests/host_tests_runner/mbed_base.py

```diff
@@ -92,10 +112,72 @@ def copy_image(self, image_path=None, disk=None, copy_method=None, port=None):
+
+            # Call proper copy method
+            result = self.copy_image_raw(image_path, disk, copy_method, port)
+            sleep(self.program_cycle_s)
+            if not result:
+                continue
+
+            if target_id:
+                bad_files = Set(['ASSERT.TXT', 'FAIL.TXT'])
+
+                # Re-try at max 5 times with 0.5 sec in delay
+                for i in range(5):
+                    # mbed_lstools.create() should be done inside the loop. Otherwise it will loop on same data.
+                    mbeds = mbed_lstools.create()
+                    mbeds_by_tid = mbeds.list_mbeds_by_targetid()   # key: target_id, value mbedls_dict()
+
+                    if target_id in mbeds_by_tid:
+                        if 'mount_point' in mbeds_by_tid[target_id] and mbeds_by_tid[target_id]['mount_point']:
+                            if not initial_remount_count is None:
+                                new_remount_count = get_remount_count(disk)
+                                if not new_remount_count is None and new_remount_count == initial_remount_count:
+                                    sleep(0.5)
+                                    continue
+
+                            common_items = []
+                            try:
+                                items = Set([x.upper() for x in os.listdir(mbeds_by_tid[target_id]['mount_point'])])
+                                common_items = bad_files.intersection(items)
+                            except OSError as e:
+                                print "Failed to enumerate disk files, retrying"
+                                continue
+
+                            for common_item in common_items:
+                                full_path = os.path.join(mbeds_by_tid[target_id]['mount_point'], common_item)
+                                print "FS_ERROR: Found %s"% (full_path)
+                                bad_file_contents = "[failed to read bad file]"
+                                try:
+                                    with open(full_path, "r") as bad_file:
+                                        bad_file_contents = bad_file.read()
+                                except IOError as error:
+                                    print "ERROR opening '%s': %s" % (full_path, error)
+                                print "FS_ERROR: Contents\n%s"% (bad_file_contents)
+                                if common_item != 'FAIL.TXT':
+                                    try:
+                                        os.remove(full_path)
+                                    except OSError as error:
+                                        print "ERROR removing '%s': %s" % (full_path, error)
+                            if common_items:
+                                result = False
+                                if 'ASSERT.TXT' in common_items:
+                                    raise Exception('ASSERT.TXT found!')
```

Comment:
Also, I forgot about these two lines: 

```
if 'ASSERT.TXT' in common_items:
    raise Exception('ASSERT.TXT found!')
```

These lines were originally added because we were debugging DAPLink issues. We shouldn't actually be raising an exception anymore, can you please remove these lines? (lines 174 and 175).